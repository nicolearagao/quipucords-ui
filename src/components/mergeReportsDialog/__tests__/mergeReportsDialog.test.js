import React from 'react';
import { MergeReportsDialog } from '../mergeReportsDialog';

describe('MergeReportsDialog Component', () => {
  it('should render a basic component', () => {
    const props = {
      show: true,
      scans: [
        { id: 1, mostRecentStatus: 'completed', mostRecentReportId: 2, name: 'lorem' },
        { id: 2, mostRecentStatus: 'pending', mostRecentReportId: 2, name: 'ipsum' }
      ]
    };

    const component = renderComponent(<MergeReportsDialog {...props} />);
    expect(component.screen.render()).toMatchSnapshot('basic');
  });

  it('should render a component, pending', () => {
    const props = {
      pending: true,
      show: true,
      scans: [
        { id: 1, mostRecentStatus: 'completed', mostRecentReportId: 2, name: 'lorem' },
        { id: 2, mostRecentStatus: 'pending', mostRecentReportId: 2, name: 'ipsum' }
      ]
    };

    const component = renderComponent(<MergeReportsDialog {...props} />);
    expect(component.screen.render().querySelector('.pf-c-modal-box__body')).toMatchSnapshot('pending');
  });
});
