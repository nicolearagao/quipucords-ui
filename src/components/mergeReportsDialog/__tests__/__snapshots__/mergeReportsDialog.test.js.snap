// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MergeReportsDialog Component should render a basic component: basic 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-2"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-0"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-1"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-0"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <header
              class="pf-c-modal-box__header"
            >
              <h4
                class="pf-c-title pf-m-md"
                data-ouia-component-id="OUIA-Generated-Title-1"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                 t(form-dialog.confirmation_title, {"context":"merge-reports"}) 
              </h4>
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-2"
            >
              <div>
                <div
                  class="merge-reports-body"
                >
                  <span
                    class="merge-reports-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="#c9190b"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M409.5,319 C406.699326,316.168774 402.882413,314.575614 398.9,314.575614 C394.917587,314.575614 391.100674,316.168774 388.3,319 L319,388.3 C316.168774,391.100674 314.575614,394.917587 314.575614,398.9 C314.575614,402.882413 316.168774,406.699326 319,409.5 L421.5,512 L319,614.5 C316.168774,617.300674 314.575614,621.117587 314.575614,625.1 C314.575614,629.082413 316.168774,632.899326 319,635.7 L388.3,705 C391.100674,707.831226 394.917587,709.424386 398.9,709.424386 C402.882413,709.424386 406.699326,707.831226 409.5,705 L512,602.5 L614.5,705 C620.367618,710.877435 629.877455,710.922082 635.8,705.1 L705.1,635.8 C707.926508,632.970191 709.505822,629.128746 709.48721,625.129169 C709.468267,621.129592 707.852954,617.303145 705,614.5 L602.5,512 L705,409.5 C707.821142,406.720523 709.417879,402.93109 709.436648,398.97079 C709.455417,395.01049 707.89467,391.206092 705.1,388.4 L635.8,319 C629.932382,313.122565 620.422545,313.077918 614.5,318.9 L512,421.5 L409.5,319 Z M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 Z M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 Z"
                      />
                    </svg>
                  </span>
                  <span>
                     t(form-dialog.confirmation_heading_merge-reports, {"context":"valid"}) 
                    <div>
                      <div>
                        <span>
                           t(form-dialog.confirmation_body_merge-reports, {"context":"valid"}) 
                        </span>
                        <ul>
                          <li>
                             t(form-dialog.confirmation_body_merge-reports, {"context":"list","name":"lorem"}) 
                          </li>
                        </ul>
                      </div>
                      <div>
                        <span>
                           t(form-dialog.confirmation_body_merge-reports, {"context":"invalid"}) 
                        </span>
                        <ul>
                          <li>
                             t(form-dialog.confirmation_body_merge-reports, {"context":"list","name":"ipsum"}) 
                          </li>
                        </ul>
                      </div>
                    </div>
                  </span>
                </div>
              </div>
            </div>
            <footer
              class="pf-c-modal-box__footer"
            >
              <button
                aria-disabled="true"
                class="pf-c-button pf-m-primary pf-m-disabled"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                disabled=""
                type="button"
              >
                 t(form-dialog.label_submit, {"context":"merge-reports"}) 
              </button>
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="OUIA-Generated-Button-secondary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                 t(form-dialog.label, {"context":"cancel"}) 
              </button>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;

exports[`MergeReportsDialog Component should render a component, pending: pending 1`] = `
<div
  class="pf-c-modal-box__body"
  id="pf-modal-part-3"
>
  <div>
    <div
      class="pf-c-empty-state quipucords-empty-state"
    >
      <div
        class="pf-c-empty-state__content"
      >
        <span
          aria-hidden="true"
          aria-label="Contents"
          aria-valuetext="Loading..."
          class="pf-c-spinner pf-m-xl pf-c-empty-state__icon"
          role="progressbar"
        >
          <span
            class="pf-c-spinner__clipper"
          >
            <span
              class="pf-c-spinner__lead-ball"
            >
              <span
                class="pf-c-spinner__tail-ball"
              />
              <h3
                class="pf-c-title pf-m-lg"
                data-ouia-component-id="OUIA-Generated-Title-3"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                 t(form-dialog.empty-state_title_merge-reports, {"context":"pending"}) 
              </h3>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
